src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationScheduleTest.kt

```diff
@@ -0,0 +1,107 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity
+
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.PENTONVILLE
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isBool
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.probationTeam
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.UUID
+
+class LocationScheduleTest {
+  private val saturday = LocalDate.of(2025, 2, 22).atStartOfDay()
+  private val sunday = LocalDate.of(2025, 2, 23).atStartOfDay()
+  private val mondayNoon = LocalDate.of(2025, 2, 24).atTime(12, 0)
+  private val mondayMidnight = LocalDate.of(2025, 2, 24).atTime(0, 0)
+
+  @DisplayName("Probation tests")
+  @Nested
+  inner class Probation {
+    @Test
+    fun `should be available for probation team on Monday at noon`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
+
+      schedule.isAvailableForProbationTeam(probationTeam(code = "PROBATION_TEAM"), mondayNoon) isBool true
+    }
+
+    @Test
+    fun `should be available for probation team on Monday out of hours`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
```

Comment:
This is only testing that the schedule reports SHARED (available) for midnight, but not that the prison opening hours of 8am - 18pm will support a meeting, as that is the higher-level availability check?

Reply:
Yes. If the config/prison setup was part of the Prison entity then we could probably handle this here at a lower level.  I don't want to have to pass it down from the service .